### DISCUSSION TITLE
2020 Retrospective (feedback)

### DISCUSSION THREAD

**Discussion by DaveSkender** [DaveSkender/Stock.Indicators#277] [Ideas]
So, you've probably seen the [Product Roadmap](https://github.com/DaveSkender/Stock.Indicators/discussions/218) and the [Backlog](https://github.com/DaveSkender/Stock.Indicators/projects/1), but what do you really want to see in the next release, or over the next year?

See #940 for the 2022 retrospective.  This discussion is locked.

**Comment by vebollu** [comment]
May be a dumb question, I did some back testing with SPY data, I used the indicators, pivot points and candlesticks to see how they performed on very next day, surprisingly the success ratio has not been passed even 52%.  So I request if you have time can you please test which indicator has more success in all these indicators/candle sticks/pivot points?

**Reply by DaveSkender** [reply]
And ruin the fun for everyone?  In all seriousness, it's hard to really narrow down which indicators are "best" given the number of argument permutations, indicator combinations, use cases, definitions of success, and changing market parameters.

**Reply by DaveSkender** [reply]
For someone trying to  [develop their trading strategy](https://www.investopedia.com/articles/trading/10/create-trading-strategies.asp), they'll typically settle on certain asset classes and styles, then start to use indicators and many other sources of information.  It wouldn't be unusual for a cryptocurrency trader to have an entirely different indicator setup than a commodities or equities trader.  The best course is to use a scientific method - start with a hypothesis, test it, learn from the results, adjust the strategy, repeat.

**Reply by DaveSkender** [reply]
A better question is, can you automate that scientific method?

**Reply by vebollu** [reply]
no, Since I have the historical data I tried to see which fared better. That's why I said it's a dumb question. The back testing Idea option has came from CmlViz, so thought that in future if we hve time we can replicate this kind of futures in our code.
https://cmlviz.com/register/trademachine-pro-cml-option-back-tester/

**Reply by DaveSkender** [reply]
It's not a dumb question, it's a great question that is really hard to answer.  I was going to suggest that you look at sites like Quantopian, but I now see that it has disappeared.  They were using [some interesting backtest engines](https://www.quantrocket.com/#backtest) to help people solve this big question.

**Reply by DaveSkender** [reply]
I'd not recommend creating a backtest engine in this library, since we want to keep a [separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns); however, that might be a good idea for a different library project.

**Reply by vebollu** [reply]
it's fine. Data is there and  I have to do just run some  sql queries. I do it on my own system when I have a time, but your app library is great. I have been in the market for a year, if you don't mind can you please give some tips for doing short term trades like 5 ema moving 13 ema kind of things?

**Reply by DaveSkender** [reply]
My personal philosophy to developing a trading strategy currently follows this logic:

1. **Do not rely on one indicator** for a trade signal.  Consider that you may have one triggering signal event, but that it is also confirmed with one or more other indicators that depict favorable conditions.
2. **Use backtesting** on multiple securities to confirm your hypothesis, but understand that changing market conditions may make backtesting an invalid test of your strategy looking forward.  Be careful of curve-fitting and reverse-engineering where your strategy works perfectly for a single security, but fails everywhere else.
3. The industry standard settings for indicators **may not be the best settings** and that many or most indicators are terrible.  Don't believe everything you read in blogs and be cautious around charting sites that share unverified user-defined indicators.
4. **Strategies often do not transfer across asset classes**.  For example, a strategy that works with equities might not work with currencies.
5. **Finding your own strategy is a journey** of personal preferences, homework, testing, and continuous improvement.

It should be noted, that my approach to developing a strategy is constantly evolving and is still a work in progress and probably will be for a very long time.

**Reply by vebollu** [reply]
ty for your advice, I like your philosophy. If you don't mind and If you know any good trader twitter accounts or subscriptions can you please share those handler names or site names so that I can follow them and will learn from them too.

**Reply by DaveSkender** [reply]
[Investopedia Academy](https://academy.investopedia.com) and [StockCharts.com ChartSchool](https://school.stockcharts.com) are good places to start.  I don’t typically follow individual people or blogs due to their highly opinionated views.

**Reply by vebollu** [reply]
thank you for your help.

**Reply by vebollu** [reply]
Hi,

Do you have any plan to implement the Vortex Indicator (VI) in near future?
https://www.investopedia.com/terms/v/vortex-indicator-vi.asp

**Reply by DaveSkender** [reply]
Vortex indicator was implemented in version 1.8.0

**Comment by ray2k** [comment]
Any chance of getting a standard dev channel indicator? Its basically a best-fit linear regression and (typically 2, sometimes 1) standard deviations up/down from the line where (in the cast of 2 deviations) something like 95% of price action tends to take place. I think this can be pieced together now between the 'Slope and Linear Regression' and 'Standard Deviation' stuff but seems less than ideal.

**Reply by DaveSkender** [reply]
Have you seen this somewhere I can reference?  We usually don't do custom indicators, but I think this is doable as an extension of Slope, much like we do for `GetSma` and `GetSmaExtended`.  Let's continue this discussion here: #241

**Reply by DaveSkender** [reply]
Standard deviation channel was implemented in version 1.9.0

**Comment by AykutCanturk** [comment]
Formations... shoulder head shoulder, rising flag, falling flag...  especially shoulder head shoulder would be very usefull.

**Reply by DaveSkender** [reply]
Yep, checkout [#220](https://github.com/DaveSkender/Stock.Indicators/discussions/220#discussioncomment-1146656), I have a running list of quite a few candlestick patterns that I'd like to implement.  cc @metathron
